Tools/machines/perlmutter-nersc/Containerfile

```diff
@@ -1,167 +1,291 @@
-      -S /tmp/c-blosc                    \
-      -B /tmp/c-blosc-build              \
-      -DCMAKE_INSTALL_PREFIX=/usr .. &&  \
-    cmake --build /tmp/c-blosc-build     \
-          --target install               \
-          -j${NJOBS} &&                  \
-    rm -rf /tmp/c-blosc*
+# CMake global options
+ENV CMAKE_GENERATOR="Ninja"
 
-# Install ADIOS2 from source
-FROM base AS adios2
+# Autotools global options
+ENV FORCE_UNSAFE_CONFIGURE=1
 
-# Ensure c-blosc is installed before ADIOS2
-COPY --from=c-blosc /usr /usr
+# Pip global options
+ENV PYTHONDONTWRITEBYTECODE=1
 
-# Verify the location of Blosc library
-RUN find /usr -name 'libblosc*'
+# Custom install location
+ENV PATH="/opt/warpx/bin:${PATH}"
+ENV LD_LIBRARY_PATH="/opt/warpx/lib:${LD_LIBRARY_PATH}"
+ENV CMAKE_PREFIX_PATH="/opt/warpx:${CMAKE_PREFIX_PATH}"
 
-# Ensure Blosc library paths are correctly configured
-ENV BLOSC_LIBRARY=/usr/lib/libblosc.so.1.21.1
-ENV BLOSC_INCLUDE_DIR=/usr/include
+# Install essential system dependencies (development)
+# Perlmutter MPICH: cray-mpich/8.1.30 based on MPICH 3.4a2
+# Ubuntu 22.04 ships MPICH 4.0, so we build from source
+RUN apt-get update           && \
+    apt-get install             \
+        -y                      \
+        --no-install-recommends \
+        autoconf                \
+        build-essential         \
+        ca-certificates         \
+        cmake                   \
+        coreutils               \
+        curl                    \
+        g++                     \
+        git                     \
+        pkg-config              \
+        python3                 \
+        python3-pip             \
+        python3-dev             \
+        python3-venv            \
+        unzip                   \
+        libblas-dev             \
+        libbz2-dev              \
+        libpng-dev              \
+        liblapack-dev           \
+        libzstd-dev             \
+        ninja-build             \
+        zlib1g-dev           && \
+    rm -rf /var/lib/apt/lists/*
 
-# Install ADIOS2
-RUN git clone -b v2.8.3 https://github.com/ornladios/ADIOS2.git /tmp/adios2 && \
-    cd /tmp/adios2 &&           \
-    cmake -S . -B build         \
-          -DADIOS2_USE_Blosc=ON \
-          -DBLOSC_LIBRARY=${BLOSC_LIBRARY}         \
-          -DBLOSC_INCLUDE_DIR=${BLOSC_INCLUDE_DIR} \
-          -DADIOS2_USE_Fortran=OFF \
-          -DADIOS2_USE_Python=OFF  \
-          -DADIOS2_USE_ZeroMQ=OFF  \
+# Install MPICH 3.4 w/ GPUdirect support from source
+#   after https://docs.nersc.gov/development/containers/shifter/how-to-use/#using-mpi-in-shifter
+# Perlmutter MPICH: cray-mpich/8.1.30 based on MPICH 3.4a2
+# TODO: install libfabric from source and depend on it (or expose where embedded libfabric gets installed)
+ARG mpich=3.4.3
+ARG mpich_prefix=mpich-$mpich
+
+RUN \
+    curl -Lo $mpich_prefix.tar.gz https://www.mpich.org/static/downloads/$mpich/$mpich_prefix.tar.gz && \
```

Comment:
If we want to run with GPU-unaware MPI to work-around the Podman-HPC MPI plugin issue, we would need to patch in this to avoid an assert in startup of AMReX, which uses this function: https://github.com/pmodels/mpich/pull/5720
https://github.com/AMReX-Codes/amrex/blob/25.11/Src/Base/AMReX_ParallelDescriptor.cpp#L1547-L1549